Source/WebInspectorUI/UserInterface/Views/SourcesNavigationSidebarPanel.js

```diff
@@ -2383,7 +2383,7 @@ WI.SourcesNavigationSidebarPanel = class SourcesNavigationSidebarPanel extends W
         }
 
         for (let target of WI.targets) {
-            if (target !== WI.pageTarget) {
+            if (target.type === WI.TargetType.Worker || target.type === WI.TargetType.ServiceWorker) {
```

Comment:
Good catch! Indeed, inspecting standalone service workers is busted now.
The cause again is `target.mainResource` being `null` at this point.

But I'm confused. 🤔 The assertion in [`WI.SourcesNavigationSidebarPanel._addWorkerTargetWithMainResource()`](https://github.com/WebKit/WebKit/blob/d98c1f9162a0f5cc180204ed11aeded85cc96795/Source/WebInspectorUI/UserInterface/Views/SourcesNavigationSidebarPanel.js#L1310) seems to be expecting this target type.

```
console.assert(target.type === WI.TargetType.Worker || target.type === WI.TargetType.ServiceWorker);
```

@dcrousso can you shine some light into what's going on?

Reply:
Removed check for `target.type === WI.TargetType.ServiceWorker` and that solves the issue for service workers.

Still interested in learning if this check is sufficient to avoid breaking other inspectable targets.
What would we miss if we used this instead?

```diff
for (let target of WI.targets) {
-            if (target !== WI.pageTarget) {
+            if (target !== WI.mainTarget) {
```

Reply:
it looks like `target.type === WI.TargetType.ServiceWorker` was erroneously added in [209373@main](https://commits.webkit.org/209373@main).  i think we can/should remove it (in both places)

i think probably the most accurate/correct thing to do here would be to `target.type === WI.TargetType.Worker` or `target instances WI.WorkerTarget`, as checking for `!== WI.pageTarget`/`!== WI.mainTarget` could cause issues in the future if more `WI.Target` are introduced